scripts-components/in-view/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ComponentInitializationError } from '@mangoweb/scripts-base'
+import { IsInView, isInView, isOutOfView, isSupported } from 'isinview'
+
+interface InViewData {
+	targets?: string
+	threshold?: number
+
+	isAboveViewClass?: string
+	isInViewClass?: string
+	isBelowViewClass?: string
+	isSeenClass?: string
+}
+
+export class InView extends Component<InViewData> {
+	static componentName = 'InView'
+
+	protected readonly targets: IsInView.Target
+	protected readonly options: Partial<IsInView.Options> = {}
+
+	protected readonly isAboveViewClass: string
+	protected readonly isInViewClass: string
+	protected readonly isBelowViewClass: string
+	protected readonly isSeenClass: string
+
+	public constructor(el: HTMLElement, data: InViewData) {
+		super(el, data)
+
+		if (!isSupported()) {
+			throw new ComponentInitializationError('InView component not supported. Add IntersectionObserver polyfill.')
+		}
+
+		this.targets = this.data.targets ? this.getChildren(this.data.targets) : this.el
+
+		if (this.data.threshold) {
+			this.options.threshold = this.data.threshold
+		}
+
+		this.isAboveViewClass = this.getProp('isAboveViewClass', 'is-aboveView')
+		this.isInViewClass = this.getProp('isInViewClass', 'is-inView')
+		this.isBelowViewClass = this.getProp('isBelowViewClass', 'is-belowView')
+		this.isSeenClass = this.getProp('isSeenClass', 'is-seen')
+	}
+
+	public init() {
+		isInView(
+			this.targets,
+			(target, details) => {
+				target.classList.add(this.isSeenClass)
+				this.togglePositionClasses(target, details)
+			},
+			this.options
+		)
+
+		isOutOfView(
+			this.targets,
+			(target, details) => {
+				this.togglePositionClasses(target, details)
+			},
+			this.options
+		)
+	}
+
+	private togglePositionClasses(target: Element, details: IsInView.Details) {
+		this.toggleClass(target, this.isAboveViewClass, details.isAboveView)
+		this.toggleClass(target, this.isInViewClass, details.isInView)
+		this.toggleClass(target, this.isBelowViewClass, details.isBelowView)
+	}
+
+	private toggleClass(target: Element, className: string, force: boolean) {
```

Comment:
`toggle` znamená přepínat přítomnost třídy a naopak (on → off, off → on). Tady se toto chování neděje a jde spíše o `force` flag, který bych pojmenoval jako `add`/`active` a celou metodu tedy potom `setPositionClass`.

Reply:
Inspirací mi bylo api pro `classList.toggle` https://developer.mozilla.org/en-US/docs/Web/API/Element/classList
a `toggleClass` v jQuery https://api.jquery.com/toggleclass/

Kdyby IE podporovalo `classList.toggle`, tak jsem použil přímo to a nedělal pomocnou metodu, která kopíruje funkcionalitu moderních prohlížečů.

Domnívám se tedy, že jsem užil zažitého pojmenování pro stejnou funkci a jiný název by byl hůře pochopitelný. S tvým komentářem trochu začínám pochybovat.

Reply:
@enzy, co si myslíš teď, když koukneš třeba na to `classList.toggle`, které má taky `force` argument?

Reply:
> When only one argument is present: Toggle the class value; i.e., if the class exists then remove it and return false, if not, then add it and return true.
> When a second argument is present: If the second argument evaluates to true, add the specified class value, and if it evaluates to false, remove it.

Tady není to prvně zmiňované chování (moje pointa), ale jenom to druhé (to o čem píšeš).

Reply:
Fér. Jsem už celkem daleko od chování `classList.toggle`. Upraveno na `updateClass`.